### ISSUE TITLE
Transition the website to GitHub

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit.org#57]
This relates to transfering responsibility for the website from me personally to the core team or webmaster. It does not relate to any design or technical issues beyond separating out stuff that is mine from stuff that is nunit's and figuring out who gets what content.

It will also relate to dealing with my own copyright on some of the material so that the nunit organization is able to use it.

**Comment by CharliePoole** [comment]
Steps to be taken - I'll add more here as they come up

**Preparation**

- [x] Move blog to NUnitSoftware

- [x] Redirect nunit.xxx/blogs to new location and delete blogs directory

- [x] Move nunitlite 1.0 directory to NUnitSoftware

- [x] Move or delete nunit-summary directory

- [x] Take a backup of all the docs

- [x] Convert all pages to static HTML

- [x] Delete or move control directory (after creating static HTML)

- [x] Delete or move PHP scripts (after creating static HTML)

- [x] Decide whether to prune the old docs, move them elsewhere or delete them

**New Site**

- [x] Move pages to GitHub

- [x] Point all the NUnit domains at GitHub initially

**Eventually**

- [ ] Point nunit.com to Charlie's site when it's ready

**Comment by CharliePoole** [comment]
@nunit/core-team The steps in the above comment seem about complete now, please review.

**Comment by CharliePoole** [comment]
@nunit/core-team I'm down to the checklist item where we decide if we are keeping all the old docs.

@rprouse wanted to keep only 2.6.4
@ChrisMaddock was for keeping them all IIRC

Looking at the pages that are accessed, I see a pretty good amount of activity as far back as NUnit 2.2.4! Which leads me to at least not want to erase them.

I will make a backup of the docs and save them.

**Comment by CharliePoole** [comment]
I updated the checklist to postpone the decision about keeping all the docs.

**Comment by jnm2** [comment]
The checked items look good to me from browsing the public site.

**Comment by jnm2** [comment]
I think we should keep all documentation with steady access. Can we tell whether they are crawlers or people?

**Comment by CharliePoole** [comment]
If we aren't changing the old docs, then there's really no downside to keeping them somewhere. 

From the reports I saw, I couldn't coordinate that info with the actual subdirectories used. I think it would need a custom report.

**Comment by ChrisMaddock** [comment]
Great that we have that info. 😄 

NuGet is another good source for this. See the package downloads from the last 6 weeks - 2.x versions still get a significant number of downloads. Worth remembering that several libraries (i.e. Xamarin UI, Unity) are currently locked to older 2.x versions of NUnit. My company was locked to NUnit 2.5.7 for years - purely because updating the version required changing filepaths all over our legacy codebase/CI structure!

https://www.nuget.org/stats/packages/NUnit?groupby=Version

My opinion hasn't changed, if it's possible to keep the old docs under the same URLs, with reasonable effort, I think that would be great.

**Comment by CharliePoole** [comment]
@ChrisMaddock I said I didn't see the downside of keeping it all, but that's a bit of an exaggeration. More accurate to say there are no significant downsides.

The potential negatives I can imagine:

* Storage space - I can't imagine that's a big deal in any environment where we would host.

* Conversion effort - that's a no-op if HTML is acceptable. I would not try to convert these to markdown, since we don't intend to update them.

* Confusion to users - that's the biggest risk, but we can mitigate it if we put the legacy docs an extra click away. Those docs are already labeled as legacy on every page.

@rprouse Were there other negatives of concern to you?

**Comment by jnm2** [comment]
Just coming back to add... I *still* blunder into NUnit 2 documentation every time  Ido a Google search for any NUnit API. I'd rather not, but it is what it is. That may be a portion of the traffic.

**Comment by CharliePoole** [comment]
We may eventually need to do some SEO.

**Comment by rprouse** [comment]
The only negatives to me are conversions from their current PHP format to HTML. Once that is done, then I don't think it is a major issue to keep them.

My second concern was that if we do keep them, I think they should maintain their URI's if possible so that we don't break incoming links. I'm not sure how we would do that if they are static pages and we have a static only site. The URI format is currently, http://nunit.org/index.php?p=equalConstraint&r=2.5.2

We could just generate the documentation pages out to versioned directories with the filenames equal to the `p` value in the query. Our documentation page `index.php` could straight HTML/JS and use Javascript to load the correct document into the DOM.

**Comment by CharliePoole** [comment]
That's a tough one! The conversion to html is trivial. We will end up with separate directories for each version, without any references to older directories for unchanged items as we now have. If PHP is available, it's pretty trivial to write a new index that forwards to the html. OTOH, if PHP is available, you can just keep it all as PHP.

I'm happy to change the plan to keep hosting it and we can set up a subdomain if you like. The legacy stuff will just keep looking old-fashioned. That may be better than doing a bunch of work to change the look of it with javascript and css. It is legacy!

**Comment by rprouse** [comment]
Before you go to the trouble of doing the HTML conversion, let me see what I can do. I was thinking of duplicating the PHP code in Javascript so we could maintain all the links. I'm not sure if it will work, but if it does, then it might be a good option.

**Comment by CharliePoole** [comment]
Just so you understand you're saving me the trouble of clicking a button. :smile:

That's for the docs anyway.

**Comment by rprouse** [comment]
Do the static docs that get generated include the header and footer? I was hoping to wrap them in the new website.

**Comment by CharliePoole** [comment]
They do. They are standalone html files. The reason the capability exists is that we originally were hosted by Thoughtworks and they didn't allow us to use any scripting. So I would update on a local machine and run a meta-script that created static html. For many years, the static html was then updated in SVN as well as being uploaded to the website.

It's not exactly how the program is organized, but you can think of the generation as doing three different things:

1. Creating multiple copies of the html, one per release, rather than a PHP include as we now have in files that didn't change from release to release.

2. Executing / expanding the PHP that is included in the html fragments.

3. Adding headers and footers.

I could drop the headers and footers easily enough.

Another approach would be to change the headers and footers of the existing site first. That only involves changing two files. Are the header and footer on github each in some file?

**Comment by rprouse** [comment]
I didn't realize it was that flexible. There isn't header and footer, but rather a layout that includes a head, header, content and footer within bare bones html.

I think if you dropped the headers and footers it would be easiest. Then we can always pull them in from the new site and we could reskin them if we ever wanted to.

If you are interested or it helps, the layout is at https://github.com/nunit/nunit.github.io/blob/master/_layouts/default.html and the includes are all in https://github.com/nunit/nunit.github.io/tree/master/_includes

**Comment by CharliePoole** [comment]
I generated the entire site as static HTML. Files look like this one:

```
html
<!-- saved from url=(0014)about:internet --><!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN">
<html>
<!-- Standard Head Part -->
<head>
<title>NUnit - DocHome</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta http-equiv="Content-Language" content="en-US">
<meta name="norton-safeweb-site-verification" content="tb6xj01p4hgo5x-8wscsmq633y11-e6nhk-bnb5d987bseanyp6p0uew-pec8j963qlzj32k5x9h3r2q7wh-vmy8bbhek5lnpp5w4p8hocouuq39e09jrkihdtaeknua" />
<link rel="stylesheet" type="text/css" href="nunit.css">
<link rel="shortcut icon" href="favicon.ico">
</head>
<!-- End Standard Head Part -->

<body>

<!-- Standard Header for NUnit.org -->
<div id="header">
  <a id="logo" href="index.html"><img src="img/logo.gif" alt="NUnit.org" title="NUnit.org"></a>
  <div id="nav">

<a href="index.html">Home</a><a href="download.html">Download</a><a href="docHome.html">Documentation</a><a href="https://github.com/nunit/docs/wiki">Wiki</a><a href="contactUs.html">Contact&nbsp;Us</a>
  </div>
</div>

<div class='notice'>NUnit 2.6.4 Legacy Documentation. View <a href='https://github.com/nunit/docs/wiki/NUnit-Documentation'>NUnit 3 Documentation</a></div>
<!-- End of Header -->

<div id="content">

<h2>NUnit 2.6.4</h2>
	
<p>This documentation covers the NUnit 2.6.4 release, 
   a bugfix release to NUnit 2.6.4. This is expected to be the last release of NUnit
   version 2, with all future work being done on NUnit 3.0.

<p>Where applicable, we have marked sections with the version in which a feature 
   first appeared.</p>

<p>If you are new to NUnit, we suggest you begin by reading the 
	<a href="getStarted.html">Getting Started</a> section of this site.
	 Those who have used earlier releases may want to begin with the 
	<a href="upgrade.html">Upgrading</a> section.</p>

<p>See the 
   <a href="releaseNotes.html">Release Notes</a>   for more information on this release.</p>

<p>All documentation is included in the release packages of NUnit. Beginning with NUnit 
2.4.2, you may choose to <a href="download.html">download</a> the documentation
separately.</p>

</div>

<!-- Submenu -->
<div id="subnav">
<ul>
<li id="current"><a href="docHome.html">NUnit 2.6.4</a></li>
<ul>
<li><a href="getStarted.html">Getting&nbsp;Started</a></li>
<li><a href="writingTests.html">Writing&nbsp;Tests</a></li>
<li><a href="runningTests.html">Running&nbsp;Tests</a></li>
<li><a href="extensibility.html">Extensibility</a></li>
<li><a href="releaseNotes.html">Release&nbsp;Notes</a></li>
<li><a href="samples.html">Samples</a></li>
<li><a href="license.html">License</a></li>
</ul>
<li><a href="&r=2.6.4.html"></a></li>
<li><a href="&r=2.6.4.html"></a></li>
</ul>
</div>
<!-- End of Submenu -->


<!-- Standard Footer for NUnit.org -->
<!-- public_html/nunit/footer.php -->
<div id="footer">
  Copyright &copy; 2002-2015 NUnit.org. All Rights Reserved.
</div>
<!-- End of Footer -->
<script src="https://google-code-prettify.googlecode.com/svn/loader/run_prettify.js?skin=desert"></script>
</body>
</html>
```

The fragment from which this was generated is

```html
<h2>NUnit <?php echo nunitRelease() ?></h2>
	
<p>This documentation covers the NUnit <?php echo nunitRelease() ?> release, 
   a bugfix release to NUnit 2.6.4. This is expected to be the last release of NUnit
   version 2, with all future work being done on NUnit 3.0.

<p>Where applicable, we have marked sections with the version in which a feature 
   first appeared.</p>

<p>If you are new to NUnit, we suggest you begin by reading the 
	<?php echo nunit_doc_link( "getStarted", "Getting Started" ) ?> section of this site.
	 Those who have used earlier releases may want to begin with the 
	<?php echo nunit_doc_link( "upgrade", "Upgrading" ) ?> section.</p>

<p>See the 
   <?php echo nunit_doc_link( "releaseNotes", "Release Notes" ) ?>
   for more information on this release.</p>

<p>All documentation is included in the release packages of NUnit. Beginning with NUnit 
2.4.2, you may choose to <a href="index.php?p=download">download</a> the documentation
separately.</p>
```

By looking at these, you may be able to better tell me what you'd like to have and where it should be for you to make use of it.

**Comment by CharliePoole** [comment]
The last three checklist items under "Preparation" are on hold, since @rprouse is proceeding with the new site instead. There's no need to convert the site to static HTML, remove old directories, etc. if it will shortly become disused.

**Comment by CharliePoole** [comment]
@rprouse When do you expect to want the DNS changed? If it's when I'm gone, I need to make sure I have certain info with me in order to log on and do it.

**Comment by rprouse** [comment]
@CharliePoole you seemed a bit hesitant about the new site and mentioned a beta period. Other than minor details it seems pretty good though and now contains all of the information from the old site. Do you want to make the switch now and rename the old site to ftp.nunit.org so the archives are still available until I find a host for them?

**Comment by CharliePoole** [comment]
I can do that tonight or tomorrow.

**Comment by CharliePoole** [comment]
Here's a plan... Assuming it lets me set it up this way...

xxx = .net, .org, .com

nunit.xxx => nunit.github.io
old.nunit.xxx => old site
downloads.nunit.xxx => old.nunit.xxx/downloads
blogs.nunit.xxx => nunitsofware.com/blogs (as now)

Note: the subdomains will also work as subdirectories, i.e. blogs.nunit.org == nunit.org/blogs

**Comment by rprouse** [comment]
Sounds good. Don't forget www.nunit.xxx 😄

**Comment by CharliePoole** [comment]
I've done a little experimenting. Seems like GitHub pages puts some limitations on what you can do. We'll see.

**Comment by rprouse** [comment]
I see that you currently have the settings set to nunit.net. Isn't `org` the primary URI?

**Comment by CharliePoole** [comment]
I'm experimenting. NUnit.net seemed like the safest place to do it.

It's disappointing to learn that GitHub will only allow me to alias one domain to it. That's a limitation they impose. My assumption had been that we could do anything that DNS would allow, but that's not the case. All our domains will have to be forwarded to one domain, rather than aliased. 😦 In the end, I'll swap in nunit.org as the primary domain.

Right now, nunit.net is working but not www.nunit.net.

**Comment by CharliePoole** [comment]
As of this AM, nunit.org, www.nunit.org and nunit.net are working for me. Please verify.

The www.nunit.net url is not working yet. Both nunit.com and www.nunit.com still point to the old site.

**Comment by jnm2** [comment]
Working here.

**Comment by CharliePoole** [comment]
Both nunit.com and www.nunit.com are working now. I made changes in DNS for nunit.net and I'm waiting for them to propagate.

**Comment by ChrisMaddock** [comment]
All are good for me - looks great. 👍

**Comment by CharliePoole** [comment]
Hmm... http://www.nunit.net is not working for me. The other five work.

**Comment by CharliePoole** [comment]
BTW remaining work on this will be transitioning the DNS somewhere else. I tried to move it to GoDaddy and wasn't getting good results with the `www` subdomains so I moved it back to AcornHost. I don't know why `www.nunit.com` works and `www.nunit.net` doesn't - I'm pretty sure I set them up exactly the same way, but I'll double check today.

It's probably better not to move the DNS until we also move the registration and that has to wait till August or beyond anyway. DNS can be provided by the host or the registrar, but GitHub doesn't provide it, so it will have to be with whatever registrar is used in the future unless we pay a service to do it.

**Comment by ChrisMaddock** [comment]
Both those addresses work for me @CharliePoole. Something cache-y, perhaps?

**Comment by jnm2** [comment]
www.nunit.net and nunit.net both working here.

**Comment by rprouse** [comment]
Working for me to. I think everything in your list is done, so I am going to close this issue.

**Comment by CharliePoole** [comment]
Fair enough. I'll create one in nunit.github.io when it's time to deal with
the domain changes.
